### PR TITLE
docs: add mermaid diagram to describe the CI flow

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -132,6 +132,36 @@ Please remember to:
 * Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.
 
 
+```mermaid
+timeline
+    title Code integration journey (CI)
+    Project setup <br> (make dev)   : Code checkout
+                                    : Virtual environment
+                                    : Dependencies
+                                    : Git pre-commit hooks
```
Comment (krneta) [inline CONTRIBUTING.md:141]: Which pre-commit hooks do you mean?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -132,6 +132,36 @@ Please remember to:
...
+                                    : Local branch
+                                    : Local changes
+                                    : Local tests
+
+    Pre-Pull Request <br> (make PR)     : Code linting
+                                        : Docs linting
+                                        : Static typing analysis
+                                        : Tests (unit|integration)
```
Comment (krneta) [inline CONTRIBUTING.md:149]: What's the difference between local tests and Test (unit) ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -132,6 +132,36 @@ Please remember to:
...
+
+    Pull Request <br> (CI checks)   : Semantic PR title check
+                                    : Related issue check
+                                    : Acknowledgment check
+                                    : Code coverage diff
+                                    : Dependencies check
+                                    : GitHub Actions security check
```
Comment (krneta) [inline CONTRIBUTING.md:156]: I would remove "security" from this.

File: CONTRIBUTING.md (hunk 2 above)
Comment (math411) [reply CONTRIBUTING.md:149]: This would be running `tox` commands whereas local testing is functional testing

**Review by krneta — APPROVED** [review]

**Pull request merged**